pytorch_lightning/trainer/optimizers.py

```diff
@@ -13,103 +13,42 @@
-        # single dictionary
-        elif isinstance(optim_conf, dict):
-            _validate_optim_conf(optim_conf)
-            optimizers = [optim_conf["optimizer"]]
-            monitor = optim_conf.get("monitor", None)
-            lr_schedulers = [optim_conf["lr_scheduler"]] if "lr_scheduler" in optim_conf else []
-        # multiple dictionaries
-        elif isinstance(optim_conf, (list, tuple)) and all(isinstance(d, dict) for d in optim_conf):
-            for opt_dict in optim_conf:
-                _validate_optim_conf(opt_dict)
-            optimizers = [opt_dict["optimizer"] for opt_dict in optim_conf]
-            scheduler_dict = (
-                lambda scheduler, opt_idx: dict(scheduler, opt_idx=opt_idx)
-                if isinstance(scheduler, dict)
-                else {"scheduler": scheduler, "opt_idx": opt_idx}
-            )
-
-            lr_schedulers = [
-                scheduler_dict(opt_dict["lr_scheduler"], opt_idx)
-                for opt_idx, opt_dict in enumerate(optim_conf)
-                if "lr_scheduler" in opt_dict
-            ]
-            optimizer_frequencies = [
-                opt_dict["frequency"] for opt_dict in optim_conf if opt_dict.get("frequency", None) is not None
-            ]
-            # assert that if frequencies are present, they are given for all optimizers
-            if optimizer_frequencies and len(optimizer_frequencies) != len(optimizers):
-                raise ValueError("A frequency must be given to each optimizer.")
-        # single list or tuple, multiple optimizer
-        elif isinstance(optim_conf, (list, tuple)) and all(isinstance(opt, Optimizer) for opt in optim_conf):
-            optimizers = list(optim_conf)
-        # unknown configuration
-        else:
-            raise MisconfigurationException(
-                "Unknown configuration for model optimizers."
-                " Output from `model.configure_optimizers()` should either be:\n"
-                " * `torch.optim.Optimizer`\n"
-                " * [`torch.optim.Optimizer`]\n"
-                " * ([`torch.optim.Optimizer`], [`torch.optim.lr_scheduler`])\n"
-                ' * {"optimizer": `torch.optim.Optimizer`, (optional) "lr_scheduler": `torch.optim.lr_scheduler`}\n'
-                ' * A list of the previously described dict format, with an optional "frequency" key (int)'
-            )
-        return optimizers, lr_schedulers, optimizer_frequencies, monitor
+        return _init_optimizers_and_lr_schedulers(pl_module)
 
     def convert_to_lightning_optimizers(self):
+        r"""
+        .. deprecated:: v1.6
+            `TrainerOptimizersMixin.convert_to_lightning_optimizers` was deprecated in v1.6 and will be removed in v1.8.
+        """
+        rank_zero_deprecation(
+            "`TrainerOptimizersMixin.convert_to_lightning_optimizers` was deprecated in v1.6 and will be removed in "
+            "v1.8."
+        )
+
         def _convert_to_lightning_optimizer(trainer, optimizer):
```

Comment:
this should also use the new implementation of `_convert_to_lightning_optimizer` to avoid these falling out of sync

Reply:
Fixed